### PR TITLE
chore: migrate from npm to yarn

### PR DESCRIPTION
**Pull request by byzyk**
* Switch to `yarn` in scripts
* Update contributing guides

### CODE REVIEW

**Comment by Legends** [comment]
I know that we have agreed to use `yarn` in the `webpack` repo, in `webpack.js.org` repo too?

**Comment by byzyk** [comment]
@Legends not sure if anyone agreed on that but the current situation is a mixture of both, which is not good. I guess we started from `npm` and then switched to `yarn` but not completely. In addition to that, we already use `yarn.lock` file, we use `yarn` for CI and just for the sake of keeping it consistent with core repo. Sounds good?

File: package.json [hunk 1]
```diff
@@ -24,10 +24,10 @@
     "node": ">=6.9"
   },
   "scripts": {
-    "start": "npm run init:generated && npm run fetch && node ./antwar.bootstrap.js develop",
+    "start": "yarn init:generated && yarn fetch && node ./antwar.bootstrap.js develop",
```
Comment (montogeek) [inline package.json]: This is a problem for new contributors, Yarn is not necessary to run build and run this repository.
Make `yarn` mandatory creates an unnecessary new barrier for new contributors.

Yes, we use `yarn.lock` for the CI, you changed CI to use Yarn, but new contributors shouldn't have in it order to contribute to the documentation.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -8,20 +8,24 @@ page please create an issue first.
...
 - Fork the **webpack.js.org** repo from [the main repository][2].
 - `git clone <your-clone-url> && cd webpack.js.org`
 
 Once you are in the project directory, run the following commands:
 
-- `npm install` to pull all dependencies.
-- `npm run build` to create a production version of the site.
-- `npm start` to develop on a local webpack-dev-server: [localhost:3000][3].
-- `npm run fetch` to retrieve external documentation/data.
+- `yarn` to pull all dependencies.
```
Comment (montogeek) [inline .github/CONTRIBUTING.md]: Add a note stating that alternatively you can use Yarn instead of npm.

File: package.json (hunk 1 above)
Comment (byzyk) [reply package.json]: Thanks for the review.

I get your concern, but currently, if you follow instructions from `CONTRIBUTING.md` step-by-step, running `npm install` is creating `package-lock.json` file which can also create confusion, for example, contributors might accidentally commit it and having two lock files it's not something we want to encourage, right?

Perhaps we can at least add `package-lock.json` and state in the guides that usage of yarn is more appreciated but not mandatory? What do you think?

Comment (montogeek) [reply package.json]: Please add `package-lock.json` to `.gitignore`, Same as webpack repo.

**Comment by byzyk** [comment]
Closing in favor of #2256

**Pull request closed without merging**